### ISSUE TITLE
error  in war file

### ISSUE THREAD

**Issue by abner01061998** [jruby/warbler#419]
#Ok, look i have been trying to generate war file for an existing rails Project with jruby
What does I do ?

Install jruby
Install Rails in Jruby
Copy my project to jruby/samples
Run jruby – S bundle install
Run jruby – S bundle –deployment
And finally
Jruby -S warble

the file is deployed succesfully but when i run into mi url i get this

org.jruby.rack.RackInitializationException: java.lang.NullPointerException
at org.jruby.rack.RackInitializationException.wrap(RackInitializationException.java:31)
at org.jruby.rack.RackApplicationFactoryDecorator.init(RackApplicationFactoryDecorator.java:104)
at org.jruby.rack.RackServletContextListener.contextInitialized(RackServletContextListener.java:50)
at org.apache.catalina.core.StandardContext.contextListenerStart(StandardContext.java:5043)
at com.sun.enterprise.web.WebModule.contextListenerStart(WebModule.java:592)
at org.apache.catalina.core.StandardContext.start(StandardContext.java:5612)
at com.sun.enterprise.web.WebModule.start(WebModule.java:540)
at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:917)
at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:900)
at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:684)
at com.sun.enterprise.web.WebContainer.loadWebModule(WebContainer.java:2044)
at com.sun.enterprise.web.WebContainer.loadWebModule(WebContainer.java:1690)
at com.sun.enterprise.web.WebApplication.start(WebApplication.java:107)
at org.glassfish.internal.data.EngineRef.start(EngineRef.java:122)
at org.glassfish.internal.data.ModuleInfo.start(ModuleInfo.java:291)
at org.glassfish.internal.data.ApplicationInfo.start(ApplicationInfo.java:352)
at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:500)
at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:219)
at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:491)
at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:540)
at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:536)
at java.security.AccessController.doPrivileged(Native Method)
at javax.security.auth.Subject.doAs(Subject.java:360)
at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:535)
at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:566)
at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:558)
at java.security.AccessController.doPrivileged(Native Method)
at javax.security.auth.Subject.doAs(Subject.java:360)

**Comment by kares** [comment]
you opened this issue against two repos but did not provide additional details in neither case :(
hoping you could provide your jruby-jars version or a reproducible case here for us ...
... also we're not seeing the cause stack here - which is weird since the stack is printed by Tomcat, please try to look if there's more details -> in the log there should be a ~ `Caused by: java.lang.NullPointerException`

its probably that you're using and old JRuby version (jruby-jars locked in Gemfile.lock) -> try updating to latest if issue still persists.

**Comment by abner01061998** [comment]
im working with 
ruby '2.3.3p222', 
jruby', :engine_version => '2.3.1'

im not using tomcat im using glassfish

**Comment by kares** [comment]
> its probably that you're using and old JRuby version (jruby-jars locked in Gemfile.lock)

that is your bundler JRuby version. seems like this is a regression in 9.1.12 (to be addressed in 9.1.13), in the mean time you can revert the jruby-jars gem to use 9.1.8.0 as suggested at https://github.com/jruby/jruby/issues/4716

you also only provided the relevant stack-trace there : 
```
Caused by: java.lang.NullPointerException
at java.lang.String.getBytes(String.java:940)
at org.jruby.ast.SymbolNode.(SymbolNode.java:54)
at org.jruby.parser.ParserSupport.asSymbol(ParserSupport.java:975)
at org.jruby.parser.RubyParser$379.execute(RubyParser.java:4286)
at org.jruby.parser.RubyParser.yyparse(RubyParser.java:1677)
at org.jruby.parser.RubyParser.yyparse(RubyParser.java:1568)
at org.jruby.parser.RubyParser.parse(RubyParser.java:5365)
at org.jruby.parser.Parser.parse(Parser.java:128)
at org.jruby.parser.Parser.parse(Parser.java:77)
at org.jruby.Ruby.parseEval(Ruby.java:2722)
```
... please try to be more thorough next time when reporting issues - we can avoid a lot of noise, work on maintenance end - which you usually get for free from open-source folks :), and you can have an explanation up faster.
